pkg/logs/config/integration_config.go

```diff
@@ -45,17 +43,18 @@ const (
 // LogsProcessingRule defines an exclusion or a masking rule to
 // be applied on log lines
 type LogsProcessingRule struct {
-	Type                    string
-	Name                    string
-	ReplacePlaceholder      string `mapstructure:"replace_placeholder"`
-	Pattern                 string
+	Type               string
+	Name               string
+	ReplacePlaceholder string `mapstructure:"replace_placeholder"`
+	Pattern            string
+	// TODO: should be moved out
```

Comment:
what exactly should be moved out and why?

Reply:
```
	Reg                     *regexp.Regexp
	ReplacePlaceholderBytes []byte
```
I'd consider moving both to avoid mixing concerns, as these are compiled and not part of the actual configuration format. `LogsSource` is a good candidate to hold them.

Reply:
ok makes sense, would you like to do it in this PR?

Reply:
No, we can do that later, this PR is already too big.